src/utils/Permissions/permissionsHandler.ts

```diff
@@ -0,0 +1,120 @@
+import { Alert, Platform } from 'react-native';
+import { check, openSettings, PERMISSIONS, request, RESULTS } from 'react-native-permissions';
+
+// tslint:disable-next-line:no-var-requires
+const alertMessagesJson = require('./alertMessages.json');
+
+type permissionsT = 'camera' | 'photo';
+
+export const alertMessages = {
+  camera: {
+    main: alertMessagesJson.permissions.alert_messages.camera.main,
+    sub: alertMessagesJson.permissions.alert_messages.camera.sub,
+  },
+  photo: {
+    main: alertMessagesJson.permissions.alert_messages.camera_roll.main,
+    sub: alertMessagesJson.permissions.alert_messages.camera_roll.sub,
+  },
+};
+
+type askForPermissionsIosT = () => Promise<boolean>;
+const askForPermissionsCameraIos: askForPermissionsIosT = async () => {
+  const status = await request(PERMISSIONS.IOS.CAMERA);
+  return status === 'granted';
+};
+
+type askForPermissionsCameraAndroidT = () => Promise<boolean>;
+const askForPermissionsCameraAndroid: askForPermissionsCameraAndroidT = async () => {
+  const status = await request(PERMISSIONS.ANDROID.CAMERA);
+  return status === 'granted';
+};
+
+const askForPermissionsLibraryIos: askForPermissionsIosT = async () => {
+  const status = await request(PERMISSIONS.IOS.PHOTO_LIBRARY);
+  return status === 'granted';
+};
+
+type handlePermissionsT = () => () => Promise<boolean>;
+
+export const handlePermissionsCamera: handlePermissionsT = () => async () => {
+  if (Platform.OS === 'ios') {
+    const hasPermissions = await handlePermissionsCameraIos()();
+    return hasPermissions;
+  } else {
+    const hasPermissions = await handlePermissionsCameraAndroid()();
+    return hasPermissions;
+  }
+};
+
+const handlePermissionsCameraIos: handlePermissionsT = () => async () => {
+  const status = await check(PERMISSIONS.IOS.CAMERA);
+  switch (status) {
+    case RESULTS.GRANTED:
+      return true;
+    case RESULTS.BLOCKED:
+      showAlert('camera');
+      return false;
+    case RESULTS.DENIED:
+      return askForPermissionsCameraIos();
+    case RESULTS.UNAVAILABLE:
```

Comment:
it's the same as a default case

Reply:
true, but in the case of the permissions statuses I would like to be as explicit as possible.